instrumentation/lettuce/lettuce-5.1/src/main/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_1/OpenTelemetryTracing.java

```diff
@@ -249,50 +252,44 @@ public OpenTelemetrySpan nextSpan(TraceContext traceContext) {
     public synchronized void finish() {
       if (span != null) {
         if (name != null) {
-          String statement =
-              (args != null && !args.isEmpty()) && !name.equals("AUTH") ? name + " " + args : name;
+          String statement = RedisCommandNormalizer.normalize(name, splitArgs());
           span.setAttribute(SemanticAttributes.DB_STATEMENT, statement);
         }
         span.end();
       }
     }
 
-    private static void fillEndpoint(Span.Builder span, OpenTelemetryEndpoint endpoint) {
-      span.setAttribute(SemanticAttributes.NET_TRANSPORT, "IP.TCP");
-      span.setAttribute(SemanticAttributes.NET_PEER_IP, endpoint.ip);
-
-      StringBuilder redisUrl = new StringBuilder("redis://");
+    private static final Pattern KEY_PATTERN = Pattern.compile("(?<wrappedKey>key<(?<key>[^>]+)>)");
 
-      if (endpoint.name != null) {
-        span.setAttribute(SemanticAttributes.NET_PEER_NAME, endpoint.name);
-        redisUrl.append(endpoint.name);
-      } else {
-        redisUrl.append(endpoint.ip);
+    // this method removes the key<...> wrapper around redis keys and splits the args string
+    private List<String> splitArgs() {
+      if (args == null || args.isEmpty()) {
+        return Collections.emptyList();
       }
 
-      if (endpoint.port != 0) {
-        span.setAttribute(SemanticAttributes.NET_PEER_PORT, (long) endpoint.port);
-        redisUrl.append(":").append(endpoint.port);
+      StringBuilder newArgs = new StringBuilder(args.length());
+
+      Matcher m = KEY_PATTERN.matcher(args);
+      int end = 0;
+      while (m.find()) {
+        int wrapperStart = m.start("wrappedKey");
+
+        newArgs.append(args, end, wrapperStart).append(m.group("key"));
+
+        end = m.end("wrappedKey");
```

Comment:
I'm not sure the `wrappedKey` named grouping helps readability here

```suggestion
        int start = m.start();

        newArgs.append(args, lastEnd, start).append(m.group("key"));

        lastEnd = m.end();
```